LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/JsonHandlers/LnsData.cs

```diff
@@ -40,5 +34,106 @@ public static class LnsData
 
         internal static readonly IJsonReader<string> VersionMessageReader =
             JsonReader.Object(JsonReader.Property("station", JsonReader.String()));
+
+
+        /*
+                  {
+                    ...
+                    "DR"    : INT,
+                    "Freq"  : INT,
+                    "upinfo": {
+                      "rctx"    : INT64,
+                      "xtime"   : INT64,
+                      "gpstime" : INT64,
+                      "rssi"    : FLOAT,
+                      "snr"     : FLOAT
+                    }
+                  }
+               */
+
+        internal static class RadioMetadataProperties
+        {
+            public static readonly IJsonProperty<DataRate> DataRate =
+                JsonReader.Property("DR", from n in JsonReader.Byte() select new DataRate(n));
+
+            public static readonly IJsonProperty<Hertz> Freq =
+                JsonReader.Property("Freq", from n in JsonReader.UInt32() select new Hertz(n));
+
+            public static readonly IJsonProperty<RadioMetadataUpInfo> UpInfo =
+                JsonReader.Property("upinfo",
+                                    JsonReader.Object(JsonReader.Property("rctx", JsonReader.UInt32()),
+                                                      JsonReader.Property("xtime", JsonReader.UInt64()),
+                                                      JsonReader.Property("gpstime", JsonReader.UInt32()),
+                                                      JsonReader.Property("rssi", JsonReader.Double()),
+                                                      JsonReader.Property("snr", JsonReader.Single()),
+                                                      (Rctx, Xtime, GpsTime, Rssi, Snr) => new RadioMetadataUpInfo(Rctx, Xtime, GpsTime, Rssi, Snr)));
```

Comment:
Please use camel casing for arguments per .NET conventions:

```suggestion
                                                      (rctx, xtime, gpsTime, rssi, snr) => new RadioMetadataUpInfo(rctx, xtime, gpsTime, rssi, snr)));
```